subprojects/problems/src/testFixtures/groovy/org/gradle/api/problems/UsesTestProblems.groovy

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.problems
+
+
+import org.gradle.api.problems.internal.DefaultProblems
+import org.gradle.internal.operations.BuildOperationProgressEventEmitter
+import org.gradle.internal.operations.OperationIdentifier
+
+/**
+ * Utility to be implemented by test classes that provides methods for creating [Problems] instances for testing.
+ */
+interface UsesTestProblems {
```

Comment:
I think there is a broad mix of interfaces, traits and static util classes used as fixtures throughout.

In this case, I was anticipating future expansion of what it would mean that a test uses the Problems API and it seemed a natural fit for an interface to me.  I don't feel very strongly about this though.